code/game/objects/machinery/miner.dm

```diff
@@ -0,0 +1,189 @@
+#define MINER_RUNNING	0
+#define MINER_SMALL_DAMAGE	1
+#define MINER_MEDIUM_DAMAGE	2
+#define MINER_DESTROYED	3
+
+///Resource generator that produces a certain material that can be repaired by marines and attacked by xenos, Intended as an objective for marines to play towards to get more req gear
+/obj/machinery/miner
+	name = "\improper Nanotrasen phoron Mining Well"
+	desc = "Top-of-the-line Nanotrasen research drill, used to extract phoron in vast quantities. Selling the phoron mined by these would net a nice profit..."
+	icon = 'icons/obj/mining_drill.dmi'
+	density = TRUE
+	icon_state = "mining_drill_active"
+	anchored = TRUE
+	resistance_flags = INDESTRUCTIBLE
+	obj_integrity = 100
+	max_integrity = 100
+	///How many sheets of material we have stored
+	var/stored_mineral = 0
+	///Current status of the miner
+	var/miner_status = MINER_RUNNING
+	///Tracks how many ticks have passed since we last added a sheet of material
+	var/add_tick = 0
+	///How many times we neeed to tick for a resource to be created, in this case this also eqates to seconds
+	var/required_ticks = 3  //make one phoron per 3 seconds.

```

Comment:
this is using SSmachines for processing which processes every 20 deciseconds so this is every 6 seconds not 3